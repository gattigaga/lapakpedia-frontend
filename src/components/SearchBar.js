import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";

const Input = styled.input`
  width: 240px;
  border: 0px;
  border-bottom: 1px solid black;
  padding: 4px 0px;
  outline: none;
  font-size: 14px;
  font-family: Roboto;
  letter-spacing: 0.1em;
`;

const SearchBar = ({ className, keyword, onChange, onPressEnter }) => (
  <Input
    type="text"
    className={className}
    value={keyword}
    onChange={onChange}
    onKeyPress={event => event.key === "Enter" && onPressEnter()}
    placeholder="Search Products"
  />
);

SearchBar.propTypes = {
  onChange: PropTypes.func,
  onPressEnter: PropTypes.func,
  keyword: PropTypes.string,
  className: PropTypes.string
};

SearchBar.defaultProps = {
  onChange: () => {},
  onPressEnter: () => {}
};

export default SearchBar;
