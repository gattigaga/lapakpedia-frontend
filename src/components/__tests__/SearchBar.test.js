import React from "react";
import { shallow } from "enzyme";
import toJSON from "enzyme-to-json";

import SearchBar from "components/SearchBar";

describe("SearchBar", () => {
  const setup = propOverrides => {
    const props = {
      ...propOverrides
    };

    const wrapper = shallow(<SearchBar {...props} />);

    return {
      props,
      wrapper
    };
  };

  it("should renders default", () => {
    const { wrapper } = setup();

    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it("should call 'onChange' callback while changed", () => {
    const onChange = jest.fn();
    const { wrapper } = setup({ onChange });
    const event = {
      target: {
        value: "MacBook Pro"
      }
    };

    wrapper.simulate("change", event);

    expect(onChange).toBeCalledWith(event);
  });

  it("should call 'onPressEnter' callback while Enter key has been pressed", () => {
    const onPressEnter = jest.fn();
    const { wrapper } = setup({ onPressEnter });
    const event = {
      key: "Enter"
    };

    wrapper.simulate("keypress", event);

    expect(onPressEnter).toBeCalled();
  });
});
