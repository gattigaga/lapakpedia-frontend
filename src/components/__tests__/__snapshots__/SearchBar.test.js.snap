// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar should renders default 1`] = `
<styled.input
  onChange={[Function]}
  onKeyPress={[Function]}
  placeholder="Search Products"
  type="text"
/>
`;

exports[`SearchBar should renders with keyword 1`] = `
<styled.input
  onChange={[Function]}
  onKeyPress={[Function]}
  placeholder="Search Products"
  type="text"
  value="MacBook Pro 2018"
/>
`;
